shopelectro/tests/tests_views.py

```diff
@@ -516,3 +519,32 @@ def test_admin_autocomplete_has_no_model_pages(self):
             f'?term={self.QUOTED_SIGNLE_RESULT_TERM}&pageType=category'
         )
         self.assertTrue(len(json_to_dict(response)) == 1)
+
+
+class Redirects(TestCase):
+
+    fixtures = ['dump.json']
+
+    # will be resurrected at rf#140
+    @unittest.expectedFailure
+    def test_redirect_on_existing_page(self):
+        """DB based redirect from existing url should do, but should not avoid it."""
```

Comment:
a test is too big and complicated. Let's divide it into smaller pieces

Reply:
@artemiy312 , it's single test case.

Do test this case, we should check two things:
`url_from returns 200` and `url_to returns 301`

Reply:
@duker33 We can test it separate. 
```python
self.redirect = Redirect.objects.create(  # also we may move it to fixtures
            site=Site.objects.first(),
            old_path=url_from,
            new_path=url_to
        )

def test_redirect_301(self):
    self.assertEqual(self.client.get(self.redirect.old_path).status_code, 300)

def test_following_redirect_200(self):
    self.assertEqual(self.client.get(self.redirect.old_path, follow=True).status_code, 200)
```

Reply:
@artemiy312 , `url_from` should not return 200 in all cases. Only for our single case.
So, this two methods have strong semantical binding. They both mean nothing in separate.

For a different test case, we should create a different test class.

Big test method's size doesn't look bad for me. It consists of simple and sequential steps

Reply:
discussed directly